src/App.tsx

```diff
@@ -78,6 +78,25 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    deleteSelectedSquares() : void {
+        for (let i = 0; i < this.state.highlights.length; i++) {
+            let addy : SquareAddress = this.state.highlights[i];
+            if (!this.state.permanentValues.includes(addy)) {
+                let newValues = JSON.parse(JSON.stringify(this.state.values));
+                if (newValues[addy]) {
+                    newValues[addy] = null;
+                }
+                /**
+                  * //TODO: CREATE marking state and transfer marking logic to board state.
+                  */
+                this.setState({
```

Comment:
Couple issues:
1. Your logic is incorrect
2. Never call setstate in a loop. Clone the state, make all modifications to the clone, then call set state once at the end.